File: Leetcode3658. GCD of odd and even sum/Solution.java
class Solution {
    public int gcdOfOddEvenSums(int n) {
        int odd_sum=0, even_sum=0;
        int gcd = 0;
        for(int i=1; i<=2*n; i++) {
            if(i%2==0)
                even_sum += i;
            else
                odd_sum += i;
        }

        while(odd_sum != 0) {
            int temp = even_sum%odd_sum;
            even_sum = odd_sum;
            odd_sum = temp;
        }
        return even_sum;
    }
}

// runtime - 0ms
return n

File: Leetcode3658. GCD of odd and even sum/Solution.py
class Solution:
    def gcdOfOddEvenSums(self, n: int) -> int:
        '''
        odd_sum=0
        even_sum=0
        for i in range(1, (2*n)+1):
            if i%2==0:
                even_sum += 1
            else:
                odd_sum += 1
        return gcd(odd_sum, even_sum)
        '''
        return n
